pkg/vendir/fetch/hg/hg.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package hg
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	ctlconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
+	ctlfetch "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/fetch"
+	ctlver "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/versions/v1alpha1"
+)
+
+type Hg struct {
+	opts       ctlconf.DirectoryContentsHg
+	infoLog    io.Writer
+	refFetcher ctlfetch.RefFetcher
+}
+
+func NewHg(opts ctlconf.DirectoryContentsHg,
+	infoLog io.Writer, refFetcher ctlfetch.RefFetcher) *Hg {
+
+	return &Hg{opts, infoLog, refFetcher}
+}
+
+// nolint:golint
+type HgInfo struct {
+	SHA         string
+	Tags        []string
```

Comment:
does not look like we are setting tags within Retrieve. should we add that or remove this field?

Reply:
This is outdated and tags are no longer present.